src/main/java/model/Customer.java

```diff
@@ -0,0 +1,27 @@
+package model;
+
+public record Customer(String firstName, String lastName, String address) {
+	public Customer {
+		validateFirstName(firstName);
+		validateLastName(lastName);
+		validateAddress(address);
+	}
+
+	private void validateFirstName(final String firstName) {
+		if (firstName == null || !"^[a-zA-Z]{1,50}$".matches(firstName)) {
```

Comment:
Sometimes you use `Pattern.matches(...)` and sometimes `"...".matches(...)`. Please stick to only one of them (preferable the first one)